CapitalForecasting/Scripts/MVP.rmd

```diff
@@ -0,0 +1,96 @@
+---
+title: "MVP"
+output: html_document
+date: '2022-06-27'
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+```
+
+Today:
+* Create an evaluation function
+* Create a train/test split function
+* Make model multivariate (How? What data would be available before hand? Should it be assumption data? Could I use predictions of the other series for the prediction of this series?)
+
+To do:
+* Make it so multiple series can easily be modeled
+* Allow users to fiddle with inputs/expected future levels of stuff
+* Ensemble forecasting
+* Time series cross-validation
+
+```{r}
+library(tidyverse)
+library(tsibble)
+library(fable)
+library(GGally)
+library(feasts)
+```
+
+```{r fake data}
+df <- read_csv("C:/Users/AxelTorbenson/OneDrive - eCapital Advisors, LLC/Documents/templates/CapitalForecasting/Data/working_capital_fake_data.csv")
+df$date <- as.Date(as.character(df$date), format="%m/%d/%Y")
+```
+```{r}
+# convert to tsibble and add working capital and current ratio
+df %>%
+  mutate(date = yearmonth(as.character(date)),
+         working_capital = rev(working_capital)) %>%
+  as_tsibble(index = date)
+
+work_cap <- df %>%
+  mutate(date = yearmonth(as.character(date)),
+         working_capital = working_capital) %>%
+  filter(row_number() <= n() - 12) %>%
+  as_tsibble(index = date)
+```
+```{r}
+ggplot(work_cap, aes(date, working_capital)) +
+  geom_line()
+```
+
+
+```{r}
+arima_fit <- work_cap %>%
+  model(auto = ARIMA(
+    working_capital,
+    stepwise = TRUE,
+    approximation = FALSE
+  ))
```

Comment:
Looks good. Would like to hear your reasoning behind enabling `setpwise = TRUE`. 

Is that because you are planning on trying out different ARIMA parameters? You might need to add the `pdq()` special to the `formula` argument in `ARIMA()` to provide the values for the search. I believe that having only the response in the formula will fit a simple linear regression between `working_capital` and `date`. 

Same question around `approximation = FALSE`. Since these values are different from the default value of the argument I believe they should be explained either in code comments or on MVP.rmd

Take a look at the [documentation](https://fable.tidyverts.org/reference/ARIMA.html).

Reply:
Sounds good. The stepwise being TRUE was to make it take less time so I could run my code quicker while troubleshooting and running multiple models. I'll add the pdq argument.